WordPress/gutenberg
# Fix incorrect targeting of block wrappers for root padding in the edi…

## What?


Fixes #46724

## Why?


In the editor, blocks are given classnames such as `wp-block`, `block-editor-block-list__block` and `block-editor-block-list__layout`, that aren't carried through to the front end. Because of this, the rules added by `useRootPaddingAwareAlignments` have to be slightly different for editor and front end. 

Problem: in the front end, there is no generic class applied to all block wrappers, such as `wp-block`, so outer block wrappers had to be targeted with `[class*="wp-block-"]:not([class*="__"])` (where the `not` part rules out any inner wrapper with a classname starting with `wp-block` such as `wp-block-cover__inner-container`). This creates a problem in the editor, because all block wrappers have a classname with a double underscore such as `block-editor-block-list__block`.
But because the `wp-block` class is also applied to all block wrappers in the editor, we can target that instead.


## Testing Instructions





1. Insert the test markup from #46724 at the root level of a site, using a theme that sets `"useRootPaddingAwareAlignments": true` in its theme.json e.g. Twenty Twenty Three.
2. Check that root padding displays correctly on the test markup, and looks the same in both editor and front end.
3. Check that nothing has changed on the original template markup and editor still matches front end.


### Testing Instructions for Keyboard


## Screenshots or screencast
